system/jlib/jstats.cpp

```diff
@@ -2837,6 +2841,29 @@ int ScopeFilter::compareDepth(unsigned depth) const
     return 0;
 }
 
+void ScopeFilter::finishedFilter()
+{
+    //If scopeTypes are provided, then this code ensure that any scopes and ids match them
+    //but that would have little benefit, and would cause complications if the only id was removed
+
+    //Some scope types can only exist at a single level.
+    if (scopeTypes.ordinality() == 1)
+    {
+        switch (scopeTypes.item(0))
+        {
+        case SSTglobal:
+            setDepth(0);
+            break;
+        case SSTworkflow:
+            setDepth(1);
+            break;
+        case SSTgraph:
+            setDepth(2);
+            break;
```

Comment:
User may set the Depth to a value less than the values calculated above.  They could specify SSTgraph but set the depth to '1'.  Of course nothing useful would be returned but this code would override user specified parameters. May be better to use the smaller of the user specified depth and optimal depth.

Reply:
I will improve it.